cmd/tailscale/cli/up.go

```diff
@@ -307,7 +307,7 @@ func runUp(ctx context.Context, args []string) error {
 		}
 	}
 
-	controlURLChanged := curPrefs.ControlURL != prefs.ControlURL
+	controlURLChanged := curPrefs.ControlURL != prefs.ControlURL || (ipn.IsLoginServerSynonym(curPrefs.ControlURL) && ipn.IsLoginServerSynonym(prefs.ControlURL))
```

Comment:
is there a missing `!` in here? this doesn't read right to me.

Also: no tests. Can you add tests to convince me this is correct? It's already table-driven so should be easy to add some more table entries.

I assume this is the part of the fix that handles the "up" by itself shouldn't do anything?

Reply:
The current tests don't seem to cover this code. But I suppose they should, I'll figure out a way to do that.